[web-animations] Rename `Animation.progress` to `Animation.overallProgress` and use a dedicated run-time flag for it

graouts opened the pull request:
#### 94cc34f9edc79784a1968732aa0ee6eeaf56a8be
<pre>
[web-animations] Rename `Animation.progress` to `Animation.overallProgress` and use a dedicated run-time flag for it
<a href="https://bugs.webkit.org/show_bug.cgi?id=283672">https://bugs.webkit.org/show_bug.cgi?id=283672</a>
<a href="https://rdar.apple.com/140533999">rdar://140533999</a>

Reviewed by Anne van Kesteren.

The `Animation.progress` property has been renamed to `Animation.overallProgress` in
<a href="https://github.com/w3c/csswg-drafts/commit/a92c32c5c351763a84430bee0cb34e29cd5cf335.">https://github.com/w3c/csswg-drafts/commit/a92c32c5c351763a84430bee0cb34e29cd5cf335.</a>
Also, this feature is not tied to Scroll-driven Animations as initially thought, so
it should be guarded by its own run-time flag.

* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/overallProgress.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/overallProgress.tentative.html: Renamed from LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/progress.tentative.html.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/progress.tentative-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/scroll-timeline-overallProgress.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/scroll-timeline-overallProgress.tentative.html: Renamed from LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/scroll-timeline-progress.tentative.html.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/scroll-timeline-progress.tentative-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/style-change-events-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/style-change-events.html:
* LayoutTests/platform/mac-wk1/TestExpectations:
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/animation/WebAnimation.cpp:
(WebCore::WebAnimation::overallProgress const):
(WebCore::WebAnimation::progress const): Deleted.
* Source/WebCore/animation/WebAnimation.h:
* Source/WebCore/animation/WebAnimation.idl:

Canonical link: <a href="https://commits.webkit.org/287065@main">https://commits.webkit.org/287065@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/147843f73216b491ff2a3434d20322bab1f6f420

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/78197 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/57242 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/31573 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/82855 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/29459 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/80330 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/66390 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/5524 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/61235 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/19155 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/81264 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/51229 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/68116 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/41552 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/48579 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/24815 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/27796 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/71340 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/69672 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/25171 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/84219 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/77433 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/5562 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/3752 "Found 2 new test failures: imported/w3c/web-platform-tests/html/browsers/the-window-object/open-close/open-features-tokenization-noreferrer.html ipc/create-media-source-with-invalid-constraints-crash.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/69459 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/5719 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/67148 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/68716 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/12718 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/10968 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/99744 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/12091 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/5511 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/8263 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/21787 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/5500 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/8932 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/7289 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -8191,6 +8191,20 @@ WebAnimationsCustomFrameRateEnabled:
+  status: testable
+  category: animation
+  humanReadableName: "Web Animations 'overallProgress' property"
+  humanReadableDescription: "Support for the 'Animation.overallProgress' property"
```
This syntax is misleading. It's not a static property after all. I usually pick one of:

* Animation's overallProgress
* Animation.prototype.overallProgress

graouts replied:
```suggestion
  humanReadableDescription: "Support for Animation's overallProgress property"
```

annevk reviewed (approved):
Modulo web-developer-facing naming this seems okay.

graouts committed:
[web-animations] Rename `Animation.progress` to `Animation.overallProgress` and use a dedicated run-time flag for it
https://bugs.webkit.org/show_bug.cgi?id=283672
rdar://140533999

Reviewed by Anne van Kesteren.

The `Animation.progress` property has been renamed to `Animation.overallProgress` in
https://github.com/w3c/csswg-drafts/commit/a92c32c5c351763a84430bee0cb34e29cd5cf335.
Also, this feature is not tied to Scroll-driven Animations as initially thought, so
it should be guarded by its own run-time flag.

* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/overallProgress.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/overallProgress.tentative.html: Renamed from LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/progress.tentative.html.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/progress.tentative-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/scroll-timeline-overallProgress.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/scroll-timeline-overallProgress.tentative.html: Renamed from LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/scroll-timeline-progress.tentative.html.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/scroll-timeline-progress.tentative-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/style-change-events-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/style-change-events.html:
* LayoutTests/platform/mac-wk1/TestExpectations:
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/animation/WebAnimation.cpp:
(WebCore::WebAnimation::overallProgress const):
(WebCore::WebAnimation::progress const): Deleted.
* Source/WebCore/animation/WebAnimation.h:
* Source/WebCore/animation/WebAnimation.idl:

Canonical link: https://commits.webkit.org/287065@main
